src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -422,15 +423,29 @@ private void validateComplaintsConfig(Course course) {
 
         // get all courses with exercises for this user
         List<Course> courses = courseService.findAllActiveWithExercisesAndLecturesForUser(user);
-        Set<Exercise> activeExercises = courses.stream().flatMap(course -> course.getExercises().stream()).collect(Collectors.toSet());
-        log.debug("          /courses/for-dashboard.findAllActiveWithExercisesForUser in " + (System.currentTimeMillis() - start) + "ms");
+        log.debug("          /courses/for-dashboard.findAllActiveWithExercisesAndLecturesForUser in " + (System.currentTimeMillis() - start) + "ms");
 
-        if (activeExercises.isEmpty()) {
+        Map<ExerciseMode, List<Exercise>> activeExercises = courses.stream().flatMap(course -> course.getExercises().stream()).collect(Collectors.groupingBy(Exercise::getMode));
+        List<Exercise> activeIndividualExercises = Optional.ofNullable(activeExercises.get(ExerciseMode.INDIVIDUAL)).orElse(List.of());
+        List<Exercise> activeTeamExercises = Optional.ofNullable(activeExercises.get(ExerciseMode.TEAM)).orElse(List.of());
+
+        if (activeIndividualExercises.isEmpty() && activeTeamExercises.isEmpty()) {
```

Comment:
Are there any cases where activeExercises is not empty but both activeIndividualExercises and activeTeamExercises are? If not, can we simplify it to:
```activeExercises.isEmpty()```? (Was like this before).

Reply:
That should work and would be more elegant but is also a bit harder to understand since `activeExercises` is not a `List` but a `Map` of exercise groups. When the map is empty, it means that there are no groups of exercises (which of course also implies that there are no exercises, so it is correct).

The two expressions would become non-equivalent if there were any other Exercise Modes apart from `INDIVIDUAL` and `TEAM`. However, in that case the method would need to be adapted anyway.

I have no strong feelings either way. Both seem acceptable to me.